src/lib/token-utils.js

```diff
@@ -0,0 +1,41 @@
+import { WEI_PRECISION } from 'src/lib/utils';
+
+/**
+ * Launches a prompt in MetaMask to add a given token as a tracked token, allowing the user to view their balance of
+ * that token at a glance from MetaMask
+ *
+ * @param {string}    address - the address of the token contract
+
+ * @param {string}    symbol - the token's ticker symbol, e.g. 'STLOS'
+ *
+ * @param {string}    image - permalink url of the token's icon
+ *
+ * @param {string}    type - Ethereum standard of the token; default is 'ERC20'
+ *
+ * @param {number}    decimals - the number of decimals constituting the token's precision, default is 18
+ *
+ *
+ * @returns {Promise<boolean>} - true iff the user granted the request to track sTLOS
+ */
+export async function promptAddToMetamask(
+    address,
+    symbol,
+    image,
+    type = 'ERC20',
+    decimals = WEI_PRECISION,
+) {
+    return window.ethereum.request({
+        method: 'wallet_watchAsset',
+        params: {
+            type,
+            options: {
+                address,
+                symbol,
+                decimals,
+                image,
+            },
+        },
+    }).catch(({ message }) => {
+        console.error(message);
```

Comment:
I wonder if that is the way we want to handle this error because it may be something that could help the user solve their issue without depending on the console of their browser.

From the [api definition](https://eips.ethereum.org/EIPS/eip-747), I'm under the impression that this could be the kind of valid notification we would want to add tot he UI:
<img width="788" alt="Screen Shot 2022-12-02 at 21 18 00" src="https://user-images.githubusercontent.com/2939354/205425258-61da4f74-a964-461c-8de4-02afa7466d24.png">

What do you think? If you agree this is a good idea, I'm happy to create another issue for this.

Reply:
mm yeah maybe we could pop a red notification banner, it definitely wouldnt hurt. I'll add that as a todo to this PR and tack it on

Reply:
@EJSG shouldn't we use the error `message` on `$q.notify` as well? 🤔 Even if its not a message as friendly as the unsupported token, I think some silent failure for a user that is not familiar with the browser console might be even more frustrating.

I also assumed the unsupported error message would come within the `message` variable. 

If you wouldn't mind double checking the `message` variable and the scenario I described, that would be awesome!

Reply:
duh... you're 100% right, will do!

Reply:
@karynemayer fixed the message, thanks again